core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java

```diff
@@ -282,15 +211,24 @@ public void parseArgs(String programName, String[] args, Object... others) {
     updateKerberosCredentials(clientConfigFile);
   }
 
+  private  ConnectionInfo cachedInfo = null;
+  private Connector cachedConnector = null;
   protected Instance cachedInstance = null;
-  protected ClientConfiguration cachedClientConfig = null;
+  private Properties cachedProps = null;
 
   synchronized public Instance getInstance() {
-    if (cachedInstance != null)
-      return cachedInstance;
-    if (mock)
-      return cachedInstance = DeprecationUtil.makeMockInstance(instance);
-    return cachedInstance = new ZooKeeperInstance(this.getClientConfiguration());
+    if (cachedInstance == null) {
+      if (mock) {
+        cachedInstance = DeprecationUtil.makeMockInstance(instance);
```

Comment:
Mock is deprecated. If we're changing this code anyway, it's probably better to go ahead and phase it out, if it isn't going to add too many complications.

Reply:
I will remove it